cli/runtime/config/cli_discovery_sources_it_test.go

```diff
@@ -0,0 +1,217 @@
+          bucket: updated-test-bucket
+          manifestPath: updated-test-manifest-path
+          annotation: two
+          required: true
+        contextType: tmc
+currentContext:
+  k8s: test-mc
+`
+	expectedConfig := `clientOptions:
+    cli:
+        discoverySources:
+            - oci:
+                name: default
+                image: "update-default-image"
+                unknown: cli-unknown
+              contextType: k8s
+            - local:
+                name: default-local
+              contextType: k8s
+            - local:
+                name: admin-local
+                path: admin
+            - oci:
+                name: new-default
+                image: new-default-image
+servers:
+    - name: test-mc
+      type: managementcluster
+      managementClusterOpts:
+        endpoint: updated-test-endpoint
+        path: updated-test-path
+        context: updated-test-context
+        annotation: one
+        required: true
+      discoverySources:
+        - gcp:
+            name: test
+            bucket: updated-test-bucket
+            manifestPath: updated-test-manifest-path
+            annotation: one
+            required: true
+          contextType: tmc
+current: test-mc
+contexts:
+    - name: test-mc
+      type: k8s
+      group: one
+      clusterOpts:
+        isManagementCluster: true
+        annotation: one
+        required: true
+        annotationStruct:
+            one: one
+        endpoint: updated-test-endpoint
+        path: updated-test-path
+        context: updated-test-context
+      discoverySources:
+        - gcp:
+            name: test
+            bucket: updated-test-bucket
+            manifestPath: updated-test-manifest-path
+            annotation: one
+            required: true
+          contextType: tmc
+        - gcp:
+            name: test-two
+            bucket: updated-test-bucket
+            manifestPath: updated-test-manifest-path
+            annotation: two
+            required: true
+          contextType: tmc
+currentContext:
+    k8s: test-mc
+`
+	return tanzuConfigBytes, expectedConfig
+}
+
+func TestCLIDiscoverySourceIntegration(t *testing.T) {
+	tanzuConfigBytes, expectedConfig := setupData()
+	f, err := os.CreateTemp("", "tanzu_config")
+	assert.Nil(t, err)
+	fmt.Println(f.Name())
+	err = os.WriteFile(f.Name(), []byte(tanzuConfigBytes), 0644)
+	assert.Nil(t, err)
+	defer func(name string) {
+		err = os.Remove(name)
+		assert.NoError(t, err)
+	}(f.Name())
+	err = os.Setenv("TANZU_CONFIG", f.Name())
+	assert.NoError(t, err)
+	// Get CLI DiscoverySources
+	sources, err := GetCLIDiscoverySources()
+	assert.Nil(t, err)
+	assert.Equal(t, 3, len(sources))
+	// Add new OCI CLI DiscoverySource
+	ds := &configapi.PluginDiscovery{
+		OCI: &configapi.OCIDiscovery{
+			Name:  "new-default",
+			Image: "new-default-image",
+		},
+	}
+	node, err := GetClientConfigNode()
+	persist, err := setCLIDiscoverySource(node, *ds)
+	//err = SetCLIDiscoverySource(*ds)
```

Comment:
extraneous?

On the same note though, why wouldn't we be testing the public SetCLIDiscoverySource, like we are doing with the other functions? I see a similar patter in the other _it_test.go files as well.

Reply:
updated to use the public methods